File: Proyecto con oracle/cliente-react-emp/src/Software.js
export class Software {

  constructor(id, codigoComputador, nombre, version, tipo, precio, fechaInstalacion, activo) {
    this.id = id;
    this.codigoComputador = codigoComputador;
    this.nombre = nombre;
    this.version = version;
    this.tipo = tipo;
    this.precio = precio;
    this.fechaInstalacion = fechaInstalacion;
    this.activo = activo;
  }

  toJson() {
    return {
      id: this.id,
      codigoComputador: this.codigoComputador,
      nombre: this.nombre,
      version: this.version,
      tipo: this.tipo,
      precio: this.precio,
      fechaInstalacion: this.fechaInstalacion,
      activo: this.activo
    };
  }

  static fromJson(data) {
    return new Software(
      data.id,
      data.codigoComputador,
      data.nombre,
      data.version,
      data.tipo,
      data.precio,
      data.fechaInstalacion,
      data.activo
    );
  }
}

File: Proyecto con oracle/microservicio-software/src/main/java/com/example/software/model/Software.java
package com.example.software.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "SOFTWARE")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Software {

    @Id
    @Column(name = "ID")
    private int id;

    @Column(name = "CODIGO_COMPUTADOR", nullable = false)
    private int codigoComputador;

    @Column(name = "NOMBRE", nullable = false, length = 150)
    private String nombre;

    @Column(name = "VERSION", nullable = false, length = 50)
    private String version;

    @Column(name = "TIPO", nullable = false, length = 80)
    private String tipo;

    @Column(name = "PRECIO", nullable = false)
    private double precio;

    @Column(name = "FECHA_INSTALACION", nullable = false)
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime fechaInstalacion;

    @Column(name = "ACTIVO", nullable = false)
    private boolean activo = true;
}
